pkg/apis/monitoring/v1/prometheus_types.go

```diff
@@ -32,6 +32,15 @@ const (
 	PrometheusKindKey = "prometheus"
 )
 
+// ScrapeProtocol represents a protocol used by Prometheus for scraping metrics.
+// Supported values are:
```

Comment:
Do we want to specify this if validation is enabled?

Reply:
By adding the kubebuilder enumeration validation, and the value can be null using the prometheus default.
```
// +kubebuilder:validation:Enum=PrometheusProto;OpenMetricsText0.0.1;OpenMetricsText1.0.0;PrometheusText0.0.4
```